### PR TITLE
Mark constructors that do not throw exceptions as noexcept

### PR DESCRIPTION
**Pull request by AtariDreams**
By ensuring we strength each move constructor with noexcept as we can, we can avoid needless exception checking and handling.

### CODE REVIEW

**Comment by SuperWig** [comment]
1. Defaulted members are already noexcept if possible, there's no need to explicitly say so. That's why none of them do.

2. Did you actually verify this? AFAIK `list`'s move constructor can throw.

**Comment by StephanTLavavej** [comment]
There appears to be an unintentional `llvm-project` submodule change.

When we closed #3278, I noted the same "many of these things actually throw" issue that @SuperWig mentioned, including specifically `deque`'s move ctor: https://github.com/microsoft/STL/pull/3278#issuecomment-1344983353

That edit appears unchanged in this PR. I'm sorry, but we don't have the review capacity to deal with PRs when our feedback is simply ignored.

File: stl/inc/functional [hunk 1]
```diff
@@ -1818,7 +1818,7 @@ public:
         return *this;
     }
 
-    move_only_function& operator=(move_only_function&& _Other) {
+    move_only_function& operator=(move_only_function&& _Other) noexcept {
```
Comment (frederick-vs-ja) [inline stl/inc/functional:1821]: Only the change in this line is reasonable to me.

However, it seems that some people in LWG/LEWG want to keep this move assignment operator non-noexcept, see https://github.com/microsoft/STL/issues/2278#issuecomment-943360927.

Comment (CaseyCarter) [reply stl/inc/functional:1821]: If anything, this should have `noexcept(true)` with a comment explaining why we don't want to strengthen it so we don't have to keep rediscovering why.

File: stl/inc/ranges [hunk 2]
```diff
@@ -512,10 +512,10 @@ namespace ranges {
         }
 
         // clang-format off
-        _Defaultabox(_Defaultabox&&) requires is_trivially_move_constructible_v<_Ty> = default;
+        _Defaultabox(_Defaultabox&&)  noexcept requires is_trivially_move_constructible_v<_Ty> = default;
         // clang-format on
 
-        constexpr _Defaultabox(_Defaultabox&& _That) : _Engaged{_That._Engaged} {
+        constexpr _Defaultabox(_Defaultabox&& _That) noexcept : _Engaged{_That._Engaged} {
```
Comment (frederick-vs-ja) [inline stl/inc/ranges:518]: Missing exception specification here is probably a bug. `_Defaultabox` is in for `join_view::_Iterator` and `lazy_split_view`, while the standard wording uses `optional` and _`non-propagating-cache`_ respectively. However, I'm not sure whether it's OK to use `_Defaultabox` in these 2 cases.

**Pull request closed without merging**